### PR TITLE
Doxygen improvements and warnings

### PR DESCRIPTION
**Pull request by white238**
Cleaning up various things in doxygen.

* Quiet warnings
* Strip root path from path in file graphs
* Add infrastructure functionality to main page

https://serac.readthedocs.io/en/docs-white238-infrastructure/doxygen/html/index.html

### CODE REVIEW

File: src/serac/coefficients/coefficient_extensions.hpp [hunk 1]
```diff
@@ -571,6 +584,13 @@ class SurfaceElementAttrCoefficient : public mfem::Coefficient {
    */
   SurfaceElementAttrCoefficient(mfem::ParMesh& mesh, mfem::Coefficient& c) : coef_(c), pmesh_(mesh) {}
 
+  /**
+   * @brief Evaluates an element attribute-based coefficient on the a boundary element
+   *
+   * @param[in] Tr The local FE transformation
```
Comment (jonwong12) [inline src/serac/coefficients/coefficient_extensions.hpp]: Thanks @white238 for fixing a lot of these.
```suggestion
   * @param[in] Tr The local surface FE transformation
```

File: src/serac/coefficients/coefficient_extensions.hpp [hunk 2]
```diff
@@ -571,6 +584,13 @@ class SurfaceElementAttrCoefficient : public mfem::Coefficient {
...
+   * @param[in] ip The current integration point
```
Comment (jonwong12) [inline src/serac/coefficients/coefficient_extensions.hpp]:
```suggestion
   * @param[in] ip The current surface element integration point
```

**Comment by white238** [comment]
This should be ready to go.  There are no more warnings for Doxygen or Sphinx (minus one weird RTD theme/Sphinx compatibility warning).

A few files were not documented at all so I did my best to fill them in.  If you have suggestions on how to improve them and/or make them more accurate, please just do a suggestion or push a commit to this branch.

File: src/serac/coefficients/coefficient_extensions.hpp [hunk 3]
```diff
@@ -26,61 +26,76 @@
  */
 namespace serac::mfem_ext {
 
+/// @brief Internal details namespace
```
Comment (joshessman-llnl) [inline src/serac/coefficients/coefficient_extensions.hpp]: Internal *implementation* details maybe?

File: src/serac/infrastructure/profiling.hpp [hunk 4]
```diff
@@ -195,11 +197,14 @@ void startCaliperRegion(const char* name);
  */
 void endCaliperRegion(const char* name);
 
+/// @brief Provides the member typedef type that names T.
```
Comment (joshessman-llnl) [inline src/serac/infrastructure/profiling.hpp:200]: Might be useful to link to https://en.cppreference.com/w/cpp/types/type_identity which also has a note on why it's useful

File: src/serac/numerics/expr_template_ops.hpp [hunk 5]
```diff
@@ -14,128 +14,188 @@
...
+ * @return Vector result
+ */
 template <typename T>
 auto operator*(serac::VectorExpr<T>&& u, const double a)
 {
   using serac::detail::ScalarMultOp;
   return serac::detail::UnaryVectorExpr<T, ScalarMultOp>(std::move(u.asDerived()), ScalarMultOp{a});
 }
 
+/// @overload
```
Comment (joshessman-llnl) [inline src/serac/numerics/expr_template_ops.hpp:55]: 👍

**Review by joshessman-llnl — APPROVED** [review]

**Review by jamiebramwell — APPROVED** [review]
Thank you for fixing this!!

**Pull request merged**